app/screens/home/channel_list/categories_list/categories/body/index.ts

```diff
@@ -76,6 +77,43 @@ export const getChannelsFromRelation = async (relations: CategoryChannelModel[]
     return Promise.all(relations.map((r) => r.channel?.fetch()));
 };
 
+const mostRecentFirst = ([channelA, myChannelA]: [ChannelModel, MyChannelModel], [channelB, myChannelB]: [ChannelModel, MyChannelModel]) => {
+    const aLastPostAt = Math.max(channelA.createAt, myChannelA.lastPostAt);
+    const bLastPostAt = Math.max(channelB.createAt, myChannelB.lastPostAt);
+
+    return bLastPostAt - aLastPostAt;
+};
+
+const sortMyChannels = ([myChannels, channels, currentUserId]: [MyChannelModel[], ChannelModel[], string]) => {
+    const myChannelsById = myChannels.reduce((result: Record<string, MyChannelModel>, c) => {
+        result[c.id] = c;
+        return result;
+    }, {});
+
+    channels.sort((channelA, channelB) => {
+        const myChannelA = myChannelsById[channelA.id];
+        const myChannelB = myChannelsById[channelB.id];
+        return mostRecentFirst([channelA, myChannelA], [channelB, myChannelB]);
+    });
+
+    // move top if own direct message exists
+    const sortedChannels = [...channels];
+    for (let i = 0; i < channels.length; i++) {
+        const channel = channels[i];
+        const teammateId = channel.type === General.DM_CHANNEL ? getUserIdFromChannelName(currentUserId, channel.name) : undefined;
+        const isOwnDirectMessage = channel.type === General.DM_CHANNEL && currentUserId === teammateId;
+
+        if (isOwnDirectMessage) {
+            // DM (you) need to be on top of the list of reads DMs
+            sortedChannels.splice(i, 1);
+            sortedChannels.unshift(channel);
+            break;
+        }
+    }
```

Comment:
This part is for sticking always your own channel (message to youself) on top without affecting any sorting strategy like in Slack.

Reply:
This would need to be approved by a PM, and I would do this in a totally different PR if/when approved by PM.

Reply:
Hey @enahum, yes I just want to put it here for informing you. I can remove it from here no problem.

Reply:
@enahum yes I wanted to show you how I did it. I will not keep this when PR is ready.